app/components/agenda/agenda-header/agenda-actions.js

```diff
@@ -333,6 +333,42 @@ export default class AgendaAgendaHeaderAgendaActions extends Component {
     this.args.didApproveAgendaitems();
   }
 
+  @action
+  async markDecisionsForSigning() {
+    const agenda = this.args.currentAgenda;
+    const agendaitems = await agenda.agendaitems;
+    const reports = await this.store.queryAll('report', {
+      'filter[decision-activity][treatment][agendaitems][:id:]': agendaitems
+        .map((x) => x.id)
+        .join(','),
+    });
```

Comment:
We do this differently in other places (edit-meeting-modal). Just query all the reports (with piece-parts and no previous version)

```
await this.store.queryAll('report', {
      'filter[:has-no:next-piece]': true,
      'filter[:has:piece-parts]': true,
      'filter[decision-activity][treatment][agendaitems][agenda][created-for][:id:]':
        this.args.meeting.id,
    });
```

This ensures we only get generated reports (not uploaded legacy docs) and in case of BIS we get the BIS report